frontend/src/pages/ActivityReport/Pages/goalsObjectives.js

```diff
@@ -133,7 +132,14 @@ const GoalsObjectives = ({
     }
   };
 
-  const onEdit = (goal, index) => {
+  const onEdit = async (goal, index) => {
+    try {
+      await setGoalAsActivelyEdited(activityReportId, goal.goalIds);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error('failed to set goal as actively edited with this error:', err);
+    }
+
```

Comment:
We need to persist the goal-is-edited state to the database when edit is clicked